### PR TITLE
Improve rawlog performance

### PR DESCRIPTION
**Pull request by ailin-nemui**

### CODE REVIEW

**Comment by dequis** [comment]
Fancy! When does this bottleneck / how to test perf difference?

**Comment by ailin-nemui** [comment]
@dwfreed complained about this when replaying his enormous znc with largish rawlog_lines

File: src/core/rawlog.c [hunk 1]
```diff
@@ -61,12 +62,8 @@ void rawlog_destroy(RAWLOG_REC *rawlog)
 /* NOTE! str must be dynamically allocated and must not be freed after! */
 static void rawlog_add(RAWLOG_REC *rawlog, char *str)
 {
-	if (rawlog->nlines < rawlog_lines || rawlog_lines <= 2)
-		rawlog->nlines++;
-	else {
-		void *tmp = rawlog->lines->data;
-		rawlog->lines = g_slist_remove(rawlog->lines,
-					       rawlog->lines->data);
+	if (rawlog->lines->length >= rawlog_lines && rawlog_lines > 0) {
```
Comment (dwfreed) [inline src/core/rawlog.c]: Maybe make this a while?  So rawlog_lines can be reduced without needing to reconnect to have it apply to existing rawlogs.

**Comment by dwfreed** [comment]
Other than that and the backwards use of the queue (which, since it's a doubly-linked list, doesn't affect performance, just prevents the use of g_queue_foreach giving the right order of lines), looks good to me

**Comment by ailin-nemui** [comment]
while loop jay nay? @dequis

**Comment by ailin-nemui** [comment]
@dwfreed do you want to push a commit to implement your suggestions?

**Comment by vague666** [comment]
Just running the patch seems to work ok and rawlog is printed out in the correct order :)
I don't have znc to test playback performance though

**Comment by ailin-nemui** [comment]
thanks, vague

**Pull request merged**